### PR TITLE
 A better fix for the calling convention madness.

### PR DESCRIPTION
**Pull request by emilio**
See the commit message for details.

### CODE REVIEW

File: src/ir/context.rs [hunk 1]
```diff
@@ -514,12 +514,19 @@ const HOST_TARGET: &'static str =
...
-    for opt in clang_args {
+    let mut args = clang_args.iter();
+    while let Some(opt) = args.next() {
         if opt.starts_with("--target=") {
             let mut split = opt.split('=');
             split.next();
             return (split.next().unwrap().to_owned(), true);
         }
+
+        if opt == "-target" {
```
Comment (fitzgen) [inline src/ir/context.rs]: should this also handle two `-`?

Comment (emilio) [reply src/ir/context.rs]:
```
$ clang --target foo
clang-6.0: error: unsupported option '--target'
```

So i think not.

**Pull request merged**